out/notebooks/spl2/installer.ts

```diff
@@ -78,48 +80,56 @@ export async function installMissingSpl2Requirements(context: ExtensionContext,
                 lspVersion = lspVersionSetting;
             }
             if (!workspace.getConfiguration().get(configKeyLspDirectory)) {
-                const localLspDefault = path.join(context.globalStorageUri.fsPath, 'spl2', 'lsp');
+                const localLspDefault = path.join(globalStoragePath, 'spl2', 'lsp');
                 await workspace.getConfiguration().update(configKeyLspDirectory, localLspDefault, true);
             }
         } catch (err) {
             reject(`Error retrieving configuration '${configKeyLspVersion}', err: ${err}`);
+            return Promise.resolve();
         }
         if (javaLoc && lspVersion) {
             // Already set up, no need to continue
             // TODO: make sure the jar files are still in the expected location
             resolve(false);
+            return Promise.resolve();
         }
         // Setup local storage directory for downloads and installs
         try {
-            makeLocalStorage(context);
+            makeLocalStorage(globalStoragePath);
         } catch (err) {
             reject(`Error creating local artifact storage for SPL2, err: ${err}`);
+            return Promise.resolve();
         }
         
         let installedLatestLsp = false;
         if (!lspVersion) {
             // If we haven't set up a Language Server version prompt use to accept terms
             // and also confirm install of java if needed
             try {
-                const accepted = await promptToDownloadLsp(!javaLoc);
-                if (!accepted) {
-                    return;
+                if (termsStatus !== TermsAcceptanceStatus.Accepted) {
+                    const accepted = await promptToDownloadLsp(!javaLoc);
+                    if (!accepted) {
+                        reject(optOutMessage);
+                        return Promise.resolve();
+                    }
                 }
                 // Remove any existing LSP artifacts first
-                const localLspDir = getLocalLspDir(context);
+                const localLspDir = getLocalLspDir(globalStoragePath);
                 fs.rmdirSync(localLspDir, { recursive: true });
-                makeLocalStorage(context); // recreate directory
+                makeLocalStorage(globalStoragePath); // recreate directory
             
-                await getLatestSpl2Release(context, progressBar);
+                await getLatestSpl2Release(globalStoragePath, progressBar);
                 installedLatestLsp = true;
             } catch (err) {
                 reject(`Error retrieving latest SPL2 release, err: ${err}`);
+                return Promise.resolve();
             }
-        } else if (!javaLoc) {
+        } else if (!javaLoc && termsStatus !== TermsAcceptanceStatus.Accepted) {
```

Comment:
Fixes and issue where we were prompting to accept terms even if you already had accepted them